### PR TITLE
Fix setElementModel resets goggle effect

### LINKED ISSUE #2441: setElementModel resets goggle effect
**Describe the bug**
Using setElementModel resets the goggle effect.

**To reproduce**
crun Camera.goggleEffect = "thermalvision"
srun me.model = 1

**Expected behaviour**
Goggle effect should be retained after changing the player model

**Version**
Multi Theft Auto v1.5.9-release-21008

### PR DESCRIPTION
**Pull request by Proxy-99**
Fixes #2441

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaCameraDefs.cpp [hunk 1]
```diff
@@ -144,6 +144,26 @@ unsigned char CLuaCameraDefs::GetCameraInterior()
     return ucInterior;
 }
 
+void CLuaCameraDefs::SetGoggleEffect(std::string CameraGoogleEffect)
```
Comment (FileEX) [inline Client/mods/deathmatch/logic/luadefs/CLuaCameraDefs.cpp]: Rename it to ``SetCameraGoggleEffect`` please

File: Client/mods/deathmatch/logic/CClientPed.cpp [hunk 2]
```diff
@@ -5210,6 +5210,7 @@ void CClientPed::Respawn(CVector* pvecPosition, bool bRestoreState, bool bCamera
             float         fTargetRotation = m_pPlayerPed->GetTargetRotation();
             unsigned char ucInterior = GetInterior();
             unsigned char ucCameraInterior = static_cast<unsigned char>(g_pGame->GetWorld()->GetCurrentArea());
+            std::string   sCameraGoogleEffect = CLuaCameraDefs::GetCameraGoggleEffect();
```
Comment (tederis) [inline Client/mods/deathmatch/logic/CClientPed.cpp]: This is bad. CLuaCameraDefs(as well as other CLua*Defs classes) is only responsible for the interaction with Lua. It must not be available from other classes.

File: Client/mods/deathmatch/logic/luadefs/CLuaCameraDefs.cpp (hunk 1 above)
Comment (tederis) [inline Client/mods/deathmatch/logic/luadefs/CLuaCameraDefs.cpp]: All functions inside `CLua*Defs` classes must be binded to Lua.

File: Client/mods/deathmatch/logic/CClientPed.cpp (hunk 2 above)
Comment (Proxy-99) [reply Client/mods/deathmatch/logic/CClientPed.cpp]: so where I move it to class CCameraSA?

Comment (tederis) [reply Client/mods/deathmatch/logic/CClientPed.cpp]: Maybe something like this?
```
bool bOldNightVision = g_pMultiplayer->IsNightVisionEnabled();
bool bOldThermalVision = g_pMultiplayer->IsThermalVisionEnabled();

...

g_pMultiplayer->SetNightVisionEnabled(bOldNightVision, true)
g_pMultiplayer->SetThermalVisionEnabled(bOldThermalVision, true)
```

**Comment by tederis** [comment]
This PR doesn't take into account a possibility of manual goggle effects change. For example, a player may put nightvision goggles on, then execute `setCameraGoggleEffect("nightvision", false)`, change a model and it will give a nightvision effect with the noise enabled, although there should be an effect without any noise.

**Comment by Proxy-99** [comment]
> This PR doesn't take into account a possibility of manual goggle effects change. For example, a player may wear nightvision goggles up, then execute `setCameraGoggleEffect("nightvision", false)`, change a model and it will give a nightvision effect with the noise enabled, although there should be an effect without any noise.

I think if we already have the effect enabled there is no need to play the sound again so should all of them be false 
see this sound played once 
[https://youtu.be/-45qz-iUuJI?si=W7YIFVIKv4h5pX4w](https://youtu.be/-45qz-iUuJI?si=W7YIFVIKv4h5pX4w)

**Review by FileEX — APPROVED** [review]

**Pull request merged**